framework/src/outputs/png/PNGOutput.C

```diff
@@ -58,34 +68,82 @@ void
+  if (processor_id() != 0)
+    mooseInfo("PNGOutput is not currently scalable.");
+
+  if (_problem_ptr->getAuxiliarySystem().hasVariable(_variable))
+  {
+    variable_number = _problem_ptr->getAuxiliarySystem().getVariable(0, _variable).number();
+    _use_aux = true;
+  }
+
+  else if (_problem_ptr->getNonlinearSystem().hasVariable(_variable))
+    variable_number = _problem_ptr->getNonlinearSystem().getVariable(0, _variable).number();
+
+  else
+    paramError("variable", "This doesn't exist.");
+
+  const std::vector<unsigned int> var_nums = {variable_number};
 
   // If we want the background to be transparent, we need a number over 1.
   if (_transparent_background)
     _out_bounds_shade = 2;
+
   // Find the values that will be used for rescaling purposes.
   calculateRescalingValues();
 
   // Set up the mesh_function
-  _mesh_function =
-      libmesh_make_unique<MeshFunction>(*_es_ptr,
-                                        _problem_ptr->getNonlinearSystem().serializedSolution(),
-                                        _problem_ptr->getNonlinearSystem().dofMap(),
-                                        var_nums);
+  if (_use_aux)
+    _mesh_function =
+        libmesh_make_unique<MeshFunction>(*_es_ptr,
+                                          _problem_ptr->getAuxiliarySystem().serializedSolution(),
+                                          _problem_ptr->getAuxiliarySystem().dofMap(),
+                                          var_nums);
+  else
+    _mesh_function =
+        libmesh_make_unique<MeshFunction>(*_es_ptr,
+                                          _problem_ptr->getNonlinearSystem().serializedSolution(),
+                                          _problem_ptr->getNonlinearSystem().dofMap(),
+                                          var_nums);
   _mesh_function->init();
 
   // Need to enable out of mesh with the given control color scaled in reverse
-  // scaling is done, this value retains it's original value.
+  // so when scaling is done, this value retains it's original value.
   _mesh_function->enable_out_of_mesh_mode(reverseScale(_out_bounds_shade));
 }
 
 // Function to find the min and max values so that all the values can be scaled between the two.
 void
 PNGOutput::calculateRescalingValues()
 {
-  // The min and max.
-  _scaling_min = _problem_ptr->getNonlinearSystem().serializedSolution().min();
-  _scaling_max = _problem_ptr->getNonlinearSystem().serializedSolution().max();
+  // The max and min.
+  // If the max value wasn't specified in the input file, find it from the system.
+  if (_pars.isParamSetByUser("max"))
+  {
+    if (_use_aux)
+      _scaling_max = _problem_ptr->getAuxiliarySystem().serializedSolution().max();
+    else
+      _scaling_max = _problem_ptr->getNonlinearSystem().serializedSolution().max();
+  }
+  else
+    _scaling_max = _max;
+
+  // If the min value wasn't specified in the input file, find it from the system.
+  if (_pars.isParamSetByUser("min"))
```

Comment:
I believe you have the logic flipped here. If it *is* set by the user, you'll want to honor their value. Same for the check above.

Reply:
Oops!